app/src/main/java/com/kms/cura/view/fragment/RegDoc_PersonalFragment.java

```diff
@@ -76,22 +86,48 @@ private void initEdittext(View myFragmentView) {
     @Override
     public void onClick(View v) {
         if (v.getId() == R.id.btnChooseDate) {
-            Dialog dateDialog = new DatePickerDialog(getActivity(),  AlertDialog.THEME_HOLO_LIGHT, myDateListener, year, month - 1, day);
+            Dialog dateDialog = new DatePickerDialog(getActivity(), AlertDialog.THEME_HOLO_LIGHT, myDateListener, year, month - 1, day);
             dateDialog.show();
         } else if (v.getId() == R.id.btnRegisterNext) {
             // check input
             if (checkInput()) {
                 // process next page
                 Bundle bundle = createBundle();
-                Fragment professional = new RegDoc_ProfessionalFragment();
+                final Fragment professional = new RegDoc_ProfessionalFragment();
                 professional.setArguments(bundle);
-                getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.Fragment_DocReg, professional).commit();
+                pDialog = new ProgressDialog(getActivity());
+                pDialog.setMessage("Loading...");
+                pDialog.setCancelable(false);
+                showProgressDialog();
+                AsyncTask<Object, Void, Void> task = new AsyncTask<Object, Void, Void>() {
+                    private Exception exception = null;
+                    @Override
+                    protected Void doInBackground(Object[] params) {
+                        try {
+                            SpecialityController.initData();
+                            FacilityController.initData();
+                            DegreeController.initData();
+                        } catch (Exception e) {
+                            exception = e;
+                        }
+                        return null;
+                    }
+
+                    @Override
+                    protected void onPostExecute(Void aVoid) {
+                        if(exception != null) {
+                            hideProgressDialog();
+                            getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.Fragment_DocReg, professional).commit();
+                        }
+                        ErrorController.showDialog(getActivity(), exception.getMessage());
```

Comment:
Not 100% sure, but I think there's a bug here. This line should be in an else statement, otherwise there will be a null pointer error. Just reading over and trying to understand the code, I'll test it thoroughly again on Monday at work, but for now it does produce a null pointer error at the line.